NeoML/include/NeoML/Dnn/Dnn.h

```diff
@@ -496,6 +499,28 @@ NEOML_API IMathEngine* GetRecommendedGpuMathEngine( size_t memoryLimit );
 
 //------------------------------------------------------------------------------------------------------------
 
+// Manages reference counts and initial states for CDnn objects, restoring configurations once all references are removed
+class NEOML_API CDnnReferenceRegister final {
+public:
+	CDnnReferenceRegister();
+	explicit CDnnReferenceRegister(CDnn* _originalDnn);
+	CDnnReferenceRegister& operator=(CDnnReferenceRegister&& other);
+
+private:
+	~CDnnReferenceRegister();
+
+	bool learningState = true; // Initial learning state of original Dnn (before creating references)
+	// For reference dnn counter = -1, else for original dnn it stores number of created reference dnns
+	int referenceCounter = 0;
+	CDnn* originalDnn = nullptr; // Pointer to the original dnn if it's reference dnn ( nullptr otherwise )
+	// Holds a copy of the original network's random number generator, used by CDnn's CRandom& reference
+	CRandom* originalRandom = nullptr;
```

Comment:
Storing raw pointer. Better replace with CPtrOwner

Reply:
We have a `FineLite` is simple wrapper on `std` for open source version, there is no class `CPtrOwner` there.
Could we leave it that way - new in constructor and delete in destructor?

Reply:
Yes